prov/efa/src/efa_msg.c

```diff
@@ -145,7 +147,11 @@ static ssize_t efa_post_recv(struct efa_ep *ep, const struct fi_msg *msg, uint64
 	ssize_t err, post_recv_err;
 	size_t i;
 
-	ewr = ofi_buf_alloc(ep->recv_wr_pool);
+	if (!msg->data) {
```

Comment:
`msg->data` can still be `NULL`/`0` even with your changes?

Reply:
It can be NULL/0 for the dgram provider

Reply:
I think we should prevent someone from setting `msg->data` when they directly use the DGRAM provider

Not sure show though. Adding `assert(msg->data == 0)` [here](https://github.com/ofiwg/libfabric/blob/main/prov/efa/src/efa_msg.c#L408) would affect both RDM and DGRAM.